src/java/fr/paris/lutece/plugins/appointment/modules/ants/web/AppointmentAnts.java

```diff
@@ -34,92 +34,143 @@
+
+	// VIEWS
+	private static final String VIEW_PREDEMANDEFORM = "predemandeForm";
+
+	// ACTIONS
+	private static final String ACTION_PRE_SEARCH = "presearch";
+
+	// PROPERTIES
+	private static final String PROPERTY_ID_PREDEMANDE_CODE_SUFFIX = "predemande_code_";
+	private static final String PROPERTY_ERROR_MESSAGE = "ants.display.fieldsErrorMessage";
+
+
+	// PARAMETERS
+	private static final String PARAMETER_CATEGORIE = "category";
+	private static final String PARAMETER_CATEGORIE_TITRES = "titres";
+	private static final String PARAMETER_DATE_TIME = "starting_date_time";
+	private static final String PARAMETER_PROPERTY_ID_FORM = "id_form";
+	private static final String PARAMETER_NB_CONSECUTIVE_SLOTS = "nb_consecutive_slots";
+	private static final String PARAMETER_PLACES_TAKED_NOMBER = "nbPlacesToTake";
+	private static final String PARAMETER_PLACES_TAKED_NOMBER_VALUE = "nbPlacesToTake";
+	private static final String PARAMETER_ANCHOR = "anchor";
+	private static final String PARAMETER_ANCHOR_VALUE = "#step3";
+
+	// MARKERS
+	private static final String MARKER_STRATING_DATE_TIME = "starting_date_time";
+	private static final String MARKER_ID_FORM = "id_form";
+	private static final String MARKER_NB_PLACES_TO_TAKE = "nbPlacesToTake";
+
+	/**
+	 * Returns the content of the page preDemandeForm.
+	 *
+	 * @param request The HTTP request
+	 * @return The view
+	 * @throws AccessDeniedException
+	 */
+	@View(value = VIEW_PREDEMANDEFORM, defaultView = true)
+	public XPage viewPreDemandeForm(HttpServletRequest request) {
+		String dateTime = request.getParameter(PARAMETER_DATE_TIME);
+		String idForm = request.getParameter(PARAMETER_PROPERTY_ID_FORM);
+		String nbPlacesToTake = request.getParameter(PARAMETER_PLACES_TAKED_NOMBER_VALUE);
+
+		Map<String, Object> model = getModel();
+		model.put(MARKER_STRATING_DATE_TIME, dateTime);
+		model.put(MARKER_ID_FORM, idForm);
+		model.put(MARKER_NB_PLACES_TO_TAKE, nbPlacesToTake);
+
+		return getXPage(TEMPLATE_PREDEMANDEFORM, request.getLocale(), model);
+	}
+
+	/**
+	 * Redirects to main carto pview
+	 *
+	 * @param request The HTTP request
+	 * @return The view
+	 */
+	@Action(value = ACTION_PRE_SEARCH)
+	public XPage presearch(HttpServletRequest request) throws IOException {
+		Integer nbPlacesToTake = Integer.parseInt(request.getParameter(PARAMETER_PLACES_TAKED_NOMBER_VALUE));
+		String fieldsErrorMessage = AppPropertiesService.getProperty(PROPERTY_ERROR_MESSAGE);
+		String dateTime = request.getParameter(PARAMETER_DATE_TIME);
+
+		List<String> predemandeCodeList = PredemandeCodeUtils.getPredemandeCodeList(request,
+				PROPERTY_ID_PREDEMANDE_CODE_SUFFIX, nbPlacesToTake);
+
+		PredemandeCodeUtils.insertCodesPredemandeOnSession(request, predemandeCodeList);
+
+		XPage redirectionXpage = new XPage();
+		String url = null;
+
+		boolean isAllCodesNotValid = PreDemandeValidationService.processPreDemandeCodes(predemandeCodeList);
```

Comment:
According to the method thas says " @return True if all predemande codes are validated and have appointments; otherwise, false." 
=> 	this variable should named "isAllCodesValid" , no ?